GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -229,6 +229,18 @@ public virtual void Shutdown()
             this.indexParsingThread.Wait();
         }
 
+        public void AllowIndexToBeParsed()
+        {
+            // Inform the index parsing thread that git has indicated the index has changed,
+            // which means the projection needs to be updated to be correct before git proceeds
+            this.externalLockReleaseRequested.Set();
+
+            this.ClearNegativePathCacheIfPollutedByGit();

```

Comment:
Do we actually want to call this here? Or wait until the git command has completed?

Reply:
Hmm. I'm not sure.  I was keeping the code sequence together but this is to clear the negative path cache that the driver keeps.  Seems like we still want this to happen whenever we update the projection.  @wilbaker what are your thoughts?

Reply:
I am only just starting to look at the changes, but my initial thought is that it would be OK to wait for the git command to complete (but functionally it doesn't hurt to call it whenever we update the projection).

The reason that we need this call is because we're blocking git from being able to create placeholders (and we don't want the blocked paths to still be in the cache after the git command completes).  If this is a projection change in the middle of a git command, VFS4G will continue to block git and so we probably don't need to clear the cache (assuming the GVFSLock is released when rebase stops due to conflicts).

Reply:
I will test turning off the placeholder creation and see what the failures are.

Reply:
I am going to leave stopping blocking placeholder creation during a git command for a future PR.  I was running into some test failures and I'm not sure if it is something that needs to change on the git side or VFSForGit side.